Change link color to increase contrast, fix #618

NickleDave opened the pull request:
This changes the link color, to increase contrast with the body text.  

## Description
Changes the css rule for the link element in general as well as for hover and visited.

## Motivation and Context
Please see discussion in #618 
including calculation of the contrast of the current link/body colors in comparison with contrast provided by this new color as suggested by an accessibility tool
https://github.com/USRSE/usrse.github.io/issues/618#issuecomment-1264184394

## Checklist:
Making the PR so we can all see how it looks as an artifact on Circle CI

- [ ] I have posted the link for the PR in the usrse slack (#website) to ask for reviewers
- [ ] I have previewed changes locally
- [ ] I have updated the README.md if necessary

cc @usrse-maintainers

NickleDave commented:
Here's a before-and-after comparison

Before (from live page, https://us-rse.org/2022-08-04-local-regional-working-group/)
![image](https://user-images.githubusercontent.com/11934090/193377908-1ce2635f-8368-46bf-85f5-d0f9b99f8a6e.png)

If I'm staring at a zoomed in screenshot like the one above, then yes I can just see the link as darker text. But if I'm scanning a whole page of text while reading, it's very hard to find the links.

After (this PR, screenshot from artifact):
![image](https://user-images.githubusercontent.com/11934090/193377886-2bdad85a-5c21-4ed0-ad01-a8903b893a7d.png)

I guess there's a question of whether this it *too* light.  
But at least I can see it now :slightly_smiling_face:

danielskatz commented:
I'm very unsure here - I agree the old one is hard to see, but the new one does seem too light

danielskatz commented:
Also, the urlcheck fail here seems correct (not a false positive)

NickleDave commented:
> I'm very unsure here - I agree the old one is hard to see, but the new one does seem too light

Should I try with this alternative color suggested by the same tool?
(The one on the right is what's currently in this PR, the one on the left is the alternative, it's less light but still much higher contrast than the color on main)

![image](https://user-images.githubusercontent.com/11934090/193647698-7cf172c3-ad6f-492a-8231-b3e15522edd2.png)

NickleDave commented:
> Also, the urlcheck fail here seems correct (not a false positive)

👍 will check/fix later

danielskatz commented:
> Should I try with this alternative color suggested by the same tool?
> (The one on the right is what's currently in this PR, the one on the left is the alternative, it's less light but still much higher contrast than the color on main)

Yes, that seems like it might be a little better.

NickleDave commented:
> Yes, that seems like it might be a little better.

👍 I will make this change so it renders and we can compare them

NickleDave commented:
Less bright color that I just changed css to use:
![image](https://user-images.githubusercontent.com/11934090/194428139-3cd671f0-b74c-43ad-9505-b791bafd06cb.png)

Original brighter color I changed the links to:
![image](https://user-images.githubusercontent.com/11934090/194428260-74672ec3-1233-4a1c-9b90-5f19265b2f18.png)

Link color before this PR:
![image](https://user-images.githubusercontent.com/11934090/194428335-492c0093-d6b0-4dc2-b4fc-2735bebda347.png)

danielskatz commented:
I do like the new one better than the previous one, and I think better than the original. Maybe this is the best we can do with shades of purple, in balancing contrasts and readability?

NickleDave commented:
Glad to hear it @danielskatz 

Adding some discussion from Slack here:
@cmaimone said
> I don't think the original bright works.  I could be convinced of the new bright, although maybe we could bold the links to increase contrast and keep them the current color?

@JeffCarver added
> I'm color blind and the current one is difficult for me to see. The less bright is better. But a bold or underline or something would help.

I am +1 for some method in addition to color that indicates a link.
I think the most conventional thing to do is underline.
E.g. see discussion here: https://webaim.org/blog/wcag-2-0-and-link-colors/
We could use bold but then anytime someone writes bold text for emphasis it could be confused with a link.

And also here:
https://usability.yale.edu/web-accessibility/articles/links#designing-links
> Giving links a different color from the surrounding text is common on the web.
> ...
> The easiest way to provide non-color link indicators is by underlining links.

I don't mean to insult anyone's intelligence by pointing out that links are usually underlined, just trying to find some resource on what's the accepted good practice so we don't spend time trying to hash it out ourselves

danielskatz commented:
underlining also seems good to me

NickleDave commented:
Pushed a commit to add underlines but now I get a 403 when I try to go to Circle-CI

danielskatz commented:
I can get to it - artifacts in https://app.circleci.com/pipelines/github/USRSE/usrse.github.io/2618/workflows/462e3b82-68fc-4c7c-b5c8-98a758bfaf91/jobs/2629/artifacts

danielskatz commented:
example image
<img width="808" alt="Screen Shot 2022-10-06 at 7 48 02 PM" src="https://user-images.githubusercontent.com/2913845/194443308-855a4af3-8300-434f-8582-d3b3953e903c.png">

NickleDave commented:
Thank you @danielskatz 

It's back for me now.  

Same page as above just in case it helps with comparison
![image](https://user-images.githubusercontent.com/11934090/194446462-9090531a-24eb-4f98-b7b2-299bc1101ab2.png)
[image](https://user-images.githubusercontent.com/11934090/194446403-830b49f0-f220-47bf-af5a-25978e1dc3c1.png)

I don't think the underlines are too overwhelming -- we probably average, like, 1+/-1 links per post I'd guess.

danielskatz commented:
this seems like a good change to go forward with for me
